ui/apps/platform/src/Containers/NetworkGraph/utils/simulatorUtils.ts

```diff
@@ -27,3 +30,102 @@ export function getDisplayYAMLFromNetworkPolicyModification(
+            const namespaces = scopeTextFormat.namespaces.join('", "');
+            return `Policies generated from the baseline of all deployments in ${namespaceText} "${namespaces}" in cluster "${cluster}"`;
+        }
+        case 'MANY_NAMESPACES':
+            return `Policies generated from the baseline of all deployments in ${scopeTextFormat.count} namespaces in cluster "${cluster}"`;
+        case 'FEW_DEPLOYMENTS': {
+            const deploymentText = pluralize('deployment', scopeTextFormat.deployments.length);
+            const deployments = scopeTextFormat.deployments.join('", "');
+            return `Policies generated from the baseline of ${deploymentText} "${deployments}" in cluster "${cluster}"`;
+        }
+        case 'MANY_DEPLOYMENTS':
+            return `Policies generated from the baseline of ${scopeTextFormat.count} deployments in cluster "${cluster}"`;
+        default:
+            return ensureExhaustive(scopeTextFormat);
+    }
+}
+
+type ScopeTextFormat =
+    | { type: 'CLUSTER_ONLY' }
+    | { type: 'FEW_NAMESPACES'; namespaces: string[] }
+    | { type: 'MANY_NAMESPACES'; count: number }
+    | { type: 'FEW_DEPLOYMENTS'; deployments: string[] }
+    | { type: 'MANY_DEPLOYMENTS'; count: number };
+
+/**
+ * Given a selected scope, return the descriptive text format used to display longer text in the UI.
+ * By encapsulating this logic we can make reuse and i16n easier in the future, as well as increase
+ * readability of the text generators that uses this function.
+ * @param scopeHierarchy The selected scope.
+ * @param abbreviationLength The number of namespaces or deployments that can be displayed before
+ *    the text is abbreviated.
+ * @returns The descriptive text format used to display longer text in the UI.
+ */
+function getScopeTextFormat(
+    scopeHierarchy: Omit<NetworkScopeHierarchy, 'remainingQuery'>,
+    abbreviationLength = 3
+): ScopeTextFormat {
+    const { namespaces, deployments } = scopeHierarchy;
+    // Cartesian product of selected namespaces and deployments
+    const namespaceDeploymentPairs = namespaces.flatMap((n) => deployments.map((d) => `${n}/${d}`));
+
+    if (namespaceDeploymentPairs.length === 0 && namespaces.length === 0) {
+        return { type: 'CLUSTER_ONLY' };
+    }
+    if (namespaceDeploymentPairs.length === 0 && namespaces.length > abbreviationLength) {
+        return { type: 'MANY_NAMESPACES', count: namespaces.length };
+    }
+    if (namespaceDeploymentPairs.length === 0 && namespaces.length <= abbreviationLength) {
+        return { type: 'FEW_NAMESPACES', namespaces };
+    }
+    if (deployments.length > 0 && namespaceDeploymentPairs.length > abbreviationLength) {
+        return { type: 'MANY_DEPLOYMENTS', count: namespaceDeploymentPairs.length };
+    }
+    if (deployments.length > 0) {
+        return { type: 'FEW_DEPLOYMENTS', deployments: namespaceDeploymentPairs };
+    }
+
+    return { type: 'CLUSTER_ONLY' };
+}
```

Comment:
This is a bit verbose for the text generation happening here, but the intent is to make the "scope text format" logic reusable without relying on a lot of string concatenation.

By duplicating the main portions of the text, and abstracting out the subject of the text (deployments, namespaces, clusters) it should make this easier to refactor for internationalization purposes down the line.